include/zephyr/sys/spsc_pbuf.h

```diff
@@ -39,20 +80,27 @@ extern "C" {
  *
  */
 struct spsc_pbuf {
-	uint32_t len;		/* Length of data[] in bytes. */
-	uint32_t wr_idx;	/* Index of the first free byte in data[] */
-	uint32_t rd_idx;	/* Index of the first valid byte in data[] */
-	uint32_t flags;		/* Flags. See @ref SPSC_PBUF_FLAGS */
-	uint8_t data[];		/* Buffer data. */
+	struct spsc_pbuf_common common;
+	union {
+		struct spsc_pbuf_ext_cache cache;
+		struct spsc_pbuf_ext_nocache nocache;
```

Comment:
yes, that is intended. In case when there are two cores with data cache and one is a producer and second is a consumer then consumer will only modify `rd_idx` so it must be put in separate cache line than things modified by the producer.